src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,155 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.InvalidRegistrationData;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final String VALID_LOGIN = "valid_login";
+    private static final String VALID_LOGIN_2 = "valid_login2";
+    private static final String VALID_LOGIN_3 = "valid_login3";
+    private static final String VALID_PASSWORD = "1234qwer";
+    private static final String ANOTHER_VALID_PASSWORD = "QWERTY12";
+    private static final int VALID_AGE = 20;
+    private static final int ANOTHER_VALID_AGE = 31;
+    private static final String INVALID_LOGIN = "abc";
+    private static final String INVALID_PASSWORD = "12qw";
+    private static final int INVALID_AGE = 10;
+
+    private static RegistrationService registrationService;
+
+    private static User validUser;
+    private static User validUser2;
+    private static User validUser3;
+    private static User validUserSameLogin;
+    private static User invalidLoginUser;
+    private static User invalidPasswordUser;
+    private static User invalidAgeUser;
+    private static User emptyUser;
+    private static User nullLoginUser;
+    private static User nullPasswordUser;
+    private static User nullAgeUser;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
```

Comment:
registrationService = new RegistrationServiceImpl(); 
should be in `@BeforeAll` method